packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -749,27 +751,58 @@ export class Project extends ProjectBase {
 
         // save SecretsManager env variables
         if (envVariable.type === BuildEnvironmentVariableType.SECRETS_MANAGER) {
-          secretsManagerSecrets.push(Stack.of(principal).formatArn({
-            service: 'secretsmanager',
-            resource: 'secret',
-            // we don't know the exact ARN of the Secret just from its name, but we can get close
-            resourceName: `${envVariable.value}-??????`,
-            sep: ':',
-          }));
+          if (Token.isUnresolved(envVariableValue)) {
```

Comment:
Is there a test for this condition? I'm not seeing it.

Reply:
Yep - those are the `can be provided as the ARN attribute of a new Secret` and `can be provided as the ARN attribute of a new Secret, followed by a JSON key` tests.

Reply:
Ah, I guess I was looking for a test where a `Fn:Split` call was expected, as we might be splitting a token. Now I see that we're assuming the input is _partially_ an unresolved token (e.g., `${secretArn}:jsonKey`).

If I understand correctly, you're supporting something like the above, but not a "full" token'ed value. So if I create a custom resource which returns back `my-secret-name:jsonKey` as a Token, then we will be passing the Token as a whole to `secretsManagerIamResources`, and the grant won't work. Am I reading that correctly now?

Reply:
Yep, you're reading it correctly.

What's the behavior that you would want to see here?

Reply:
Frankly, I'm not sure. I don't know that there's a good solution here. I'm hoping this is a relatively edge-of-the-corner case. :D